alacritty/src/display.rs

```diff
@@ -389,6 +392,9 @@ impl Display {
         let cell_width = self.size_info.cell_width;
         let cell_height = self.size_info.cell_height;
 
+        // Update the window's min inner size.
+        self.window.set_min_inner_size(Some(min_window_size(cell_width, cell_height)));
```

Comment:
You should take padding from the config into account. It's also should be `cell_width * MIN_COLS` and `cell_height * MIN_LINES`.